lib/server.js

```diff
@@ -1133,11 +1134,24 @@ function _finishReqResCycle(req, res, route, err) {
     // regardless of res status.
 
     // after event has signature of function(req, res, route, err) {...}
-    if (!res.finished && !req.clientClosed) {
+    if (!res.finished && !req._connectionState) {
```

Comment:
Should probably explicitly check for `_connectionState === abort || closed` since when and if we add new states this might break.